seaweedfs/src/main/java/site/ycsb/db/SeaweedClient.java

```diff
@@ -0,0 +1,301 @@
+    if (!filerClient.rm(this.folder + "/" + tableName + "/" + key, true, true)) {
+      return Status.ERROR;
+    }
+    return Status.OK;
+  }
+
+  /**
+   * Initialize any state for the storage.
+   * Called once per SeaweedFS instance; If the client is not null it is re-used.
+   */
+  @Override
+  public void init() throws DBException {
+    filerHost = getProperties().getProperty("seaweed.filerHost", "localhost");
+    filerPort = Integer.parseInt(getProperties().getProperty("seaweed.filerPort", "8888"));
+    folder = getProperties().getProperty("seaweed.folder", "/ycsb");
+    filerGrpcClient = new FilerGrpcClient(filerHost, filerPort+10000);
+    filerClient = new FilerClient(filerGrpcClient);
+    filerClient.mkdirs(this.folder, 0755);
+  }
+
+  /**
+   * Create a new File in the table. Any field/value pairs in the specified
+   * values HashMap will be written into the file with the specified record
+   * key.
+   *
+   * @param tableName The name of the table
+   * @param key  The record key of the file to insert.
+   * @param values A HashMap of field/value pairs to insert in the file.
+   *         Only the content of the first field is written to a byteArray
+   *         multiplied by the number of field. In this way the size
+   *         of the file to upload is determined by the fieldlength
+   *         and fieldcount parameters.
+   * @return OK on success, ERROR otherwise. See the
+   * {@link DB} class's description for a discussion of error codes.
+   */
+  @Override
+  public Status insert(String tableName, String key,
+             Map<String, ByteIterator> values) {
+    return writeToStorage(tableName, key, values);
+  }
+
+  /**
+   * Read a file from the table. Each field/value pair from the result
+   * will be stored in a HashMap.
+   *
+   * @param tableName The name of the table
+   * @param key  The record key of the file to read.
+   * @param fields The list of fields to read, or null for all of them,
+   *         it is null by default
+   * @param result A HashMap of field/value pairs for the result
+   * @return OK on success, ERROR otherwise.
+   */
+  @Override
+  public Status read(String tableName, String key, Set<String> fields,
+             Map<String, ByteIterator> result) {
+    return readFromStorage(tableName, key, fields, result);
+  }
+
+  /**
+   * Update a file in the table. Any field/value pairs in the specified
+   * values HashMap will be written into the file with the specified file
+   * key, overwriting any existing values with the same field name.
+   *
+   * @param tableName The name of the table
+   * @param key  The file key of the file to write.
+   * @param values A HashMap of field/value pairs to update in the record
+   * @return OK on success, ERORR otherwise.
+   */
+  @Override
+  public Status update(String tableName, String key,
+             Map<String, ByteIterator> values) {
+    return writeToStorage(tableName, key, values);
```

Comment:
are the backend `SeaweedWrite.writeData` and/or `SeaweedRead.read` calls going to take care of coalescing with existing values?

Reply:
No. I added logic to merge the new value with existing value.